### ISSUE TITLE
Issues in obtaining training/validation statistics (orca pytorch)

### ISSUE THREAD

**Issue by xiejinglei** [intel/analytics-zoo#380]
In orca pytorch [notebook 6.2](https://github.com/xiejinglei/zoo-tutorials/blob/pytorch6.2/orca/pytorch/6.2-test.ipynb) , I need to collect training accuracy / loss and validation accuracy / loss in each epoch. However, orca pytorch does not save the training accuracy automatically. So I have to use a loop to train the model (1 epoch in each loop), and obtain the training accuracy using `evaluate`. It would be really inconvenient if the training set is huge. In my case, the code looks like:

```py
from zoo.orca.learn.pytorch import Estimator
from zoo.orca.learn.trigger import EveryEpoch

est = Estimator.from_torch(model=model, optimizer=rmsprop, loss=criterion, metrics=metrics)

num_epochs = 3
train_acc = []
train_loss = []
val_acc = []
val_loss = []
for i in range(1, num_epochs + 1):
    est.set_tensorboard("./log/", "epoch_" + str(i))
    print("\nfit ", i, "start\n")
    est.fit(data=train_loader, epochs=i, validation_data=val_loader, batch_size=batch_size, 
            checkpoint_trigger=EveryEpoch())
    print("\nfit ", i, "end\n")

    print("Get training accuracy: ")
    train_acc_tmp = est.evaluate(data=train_loader, batch_size=batch_size)
    train_acc.append(train_acc_tmp["Top1Accuracy"])
    train_loss_tmp = est.get_train_summary("Loss")
    train_loss.append(train_loss_tmp)

    print("Get validation accuracy: ")
    val_acc_tmp = est.get_validation_summary("Top1Accuracy")
    val_acc.append(val_acc_tmp)
    val_loss_tmp = est.get_validation_summary("Loss")
    val_loss.append(val_loss_tmp)
```

Also, it seems that the validation loss cannot be provided by the [get_validation_summary](https://github.com/intel-analytics/analytics-zoo/blob/master/pyzoo/zoo/orca/learn/pytorch/estimator.py#L520) function as well. If I print the `val_loss` in the above code, it outputs `[[], [], []]`.

Another question I am confused about is how can I obtain the right value I want using `get_validation_summary`.  For example, I would like to know the validation accuracy of the first epoch, and the `val_acc_tmp =  est.get_validation_summary("Top1Accuracy")` in the first loop gives the following data:

[[157.0, 0.5914000272750854, 1615187402.143], 
[157.0, 0.6146500110626221, 1615187422.42], 
[157.0, 0.6115999817848206, 1614934791.22], 
[157.0, 0.6050500273704529, 1614934811.794], 
[157.0, 0.6050500273704529, 1614934876.919], 
[157.0, 0.5860000252723694, 1615186397.862], 
[157.0, 0.6051999926567078, 1615186416.948], 
[157.0, 0.6308000087738037, 1615196092.246], 
[157.0, 0.646049976348877, 1615196112.892], 
[157.0, 0.5934000015258789, 1615185085.106], 
[157.0, 0.6161999702453613, 1615185105.352], 
[157.0, 0.6161999702453613, 1615185168.485], 
[157.0, 0.6161999702453613, 1615185230.499], 
[157.0, 0.5881999731063843, 1615193047.818], 
[157.0, 0.620199978351593, 1615193069.241], 
[157.0, 0.620199978351593, 1615193134.826], 
[157.0, 0.5604000091552734, 1615192203.006], 
[157.0, 0.603950023651123, 1615192223.501], 
[157.0, 0.603950023651123, 1615192291.882], 
[157.0, 0.603950023651123, 1615192361.084], 
[157.0, `0.6164000034332275`, 1615196573.049],                            ### 0.6164 is here
[157.0, 0.6290000081062317, 1615196593.948], 
[157.0, 0.6251999735832214, 1615195923.748], 
[157.0, 0.6346499919891357, 1615195945.111], 
[157.0, 0.5541999936103821, 1615188972.912], 
[157.0, 0.5728499889373779, 1615188992.813], 
[157.0, 0.6141999959945679, 1615195122.619], 
[157.0, 0.6345000267028809, 1615195142.088], 
[157.0, 0.6353999972343445, 1615186158.859], 
[157.0, 0.6522499918937683, 1615186179.808], 
[157.0, 0.6208000183105469, 1615191220.437], 
[157.0, 0.6399499773979187, 1615191240.648], 
[314.0, 0.6697999835014343, 1615196204.618], 
[314.0, 0.6828500032424927, 1615196225.974], 
[314.0, 0.6620000004768372, 1615195221.975], 
[314.0, 0.6978999972343445, 1615195241.104], 
[471.0, 0.614799976348877, 1615196317.944], 
[471.0, 0.6287999749183655, 1615196338.666], 
[471.0, 0.6588000059127808, 1615195326.895], 
[471.0, 0.687749981880188, 1615195346.983]]

But the training output is 
```py
2021-03-08 17:42:53 INFO  DistriOptimizer$:177 - [Epoch 1 20096/20096][Iteration 157][Wall Clock 42.1312477s] validate model throughput is 3405.161 records/second
2021-03-08 17:42:53 INFO  DistriOptimizer$:180 - [Epoch 1 20096/20096][Iteration 157][Wall Clock 42.1312477s] Top1Accuracy is Accuracy(correct: 3082, count: 5000, accuracy: 0.6164)
```
This means that the validation accuracy here should be  0.6164, which is an entry in the middle of the above list. Is comparing timestamp the only way to get the validation accuracy I want? I'm afraid that it may not always be a stable solution...

Thanks for your patience!
